src/services/order.service.ts

```diff
@@ -0,0 +1,57 @@
+import axios, { AxiosResponse, CancelTokenSource } from 'axios';
+
+import { CreateOrderRequest } from '../types/orders/create-order-request.type';
+import { ERROR_REQUEST_CANCELLED_BY_CLIENT } from '../constants/api.constants';
+import { IOrder } from '../interfaces/order.interface';
+import axiosInstance from '../api/axios-instance';
+import { handleApiError } from '../utils/request.utils';
+import log from '../utils/log.utils';
+
+class OrderService {
+  private readonly ORDERS_FEATURE_URL_PREFIX = '/orders';
+
+  public async placeOrder(
+    orderData: CreateOrderRequest,
+    cancelToken: CancelTokenSource
+  ) {
+    try {
```

Comment:
**suggestion:** nice work with the request cancelation mechanism (tho you overcomplicated again :)) ). As a suggestion I once again would ask you to take care for repeated implementation (e.g.: same try catch etc)

Reply:
You're absolutely right, I'm still working on better determining when should I generalize such functionality as it makes it easier to maintain, or leave it like this as it makes it more obscure. This is, as you pointed out, an example of the first case  😄 . Working on it right now. Thank you